### ISSUE TITLE
hacking test error，'NetworkModule' object has no attribute '_name'

### ISSUE THREAD

**Issue by justbiaoliu** [ansible/ansible#18697]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

lib/ansible/module_utils/basic
##### ANSIBLE VERSION

```
ansible 2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

NO configuration changes

##### OS / ENVIRONMENT

Ubuntu 14.04 LTS

##### SUMMARY

make hacking test network module base of lib/ansible/module_utils/basic.py，hacking test does not work well，error with: 'NetworkModule' object has no attribute '_name'

##### STEPS TO REPRODUCE

```
root@localhost:/usr/ansible# source ./hacking/env-setup
root@localhost:/usr/ansible# ./hacking/test-module -m ./lib/ansible/moudles/core/network/cloudengine/ce_vlan.py -a "vlan_id=1 state=absent host=*** port *** username=*** password=***"
note: cloudengine/ce_vlan.py is our network test module, you can use other network module to reproduce instead, i.e. Cisco's nxos_config.py or nxos_vlan.py module
```

##### EXPECTED RESULTS

hacking test network module base of lib/ansible/module_utils/basic.py work well
##### ACTUAL RESULTS

```
root@localhost:/usr/ansible# ./hacking/test-module -m ./lib/ansible/moudles/core/network/cloudengine/ce_vlan.py -a "vlan_id=1 state=absent host=*** port *** username=*** password=***"
* including generated source, if any, saving to: /root/.ansible_module_generated
* ansiballz module detected; extracted module source to: /root/debug_dir
************************
RAW OUTPUT

Traceback (most recent call last):
  File "/root/debug_dir/ansible_module_ce_vlan.py", line 737, in <module>
    main()
  File "/root/debug_dir/ansible_module_ce_vlan.py", line 729, in main
    vlancfg = Vlan(argument_spec)
  File "/root/debug_dir/ansible_module_ce_vlan.py", line 265, in __init__
    setf.init_module()
  File "/root/debug_dri/ansible_module_ce_vlan.py", line 299, in init_module
    argument_spec=self.spec, supports_check_mode=True) 
  File "/root/debug_dir/ansible/module_utils/network.py", line 111, in __init__
    self.connect()
  File "/root/debug_dir/ansible/module_utils/network.py", line 154, in connect
    self.params['port'], self.params['transport']))
  File "/root/debug_dir/ansible/module_utils/basic.py", line 1630, in log
    module = 'ansible-%s' % self._name
AttributeError: 'NetworkModule' object has no attribute '_name'
```

**Comment by kenwmitchell** [comment]
I'm on _2.2.0.0_ and using a custom module basically just like the _ios_facts_ module. Running with ```ansible``` or ```ansible-playbook``` works fine. Adding ```-a "_ansible_module_name=whatever"``` to ```test-module``` also works.

This didn't happen in 5f12731797baea35701b20d7d87ab22d1f4adf33

**Comment by gundalow** [comment]
I don't see any reference to `NetworkMoude` in git history
`git grep NetworkMoude $(git rev-list --all -- lib/ansible/module_utils/)`

Do you have a typo of `NetworkModule` in your code?

Could you please do: 

`grep -r NetworkMoude /path/to/ansible/`

needs_info

**Comment by kenwmitchell** [comment]
I assumed NetworkMoude was a typo. I'm using NetworkModule from
module_utils/network.

**Comment by gundalow** [comment]
On a broken checkout can you please do

`grep -r NetworkMoude /path/to/ansible/`

**Comment by kenwmitchell** [comment]
If you are asking me, of course NetworkMoude doesn't exist in my checkout. I assumed the OP mistakenly typed NetworkMoude or that NetworkMoude is a child of the NetworkModule class since the stack trace lines up with module_utils/network.NetworkModule.

This is exactly the issue I'm seeing (except with NetworkModule instead of NetworkMoude). Should I create a new issue?

**Comment by gundalow** [comment]
@kenwmitchell Please raise a fresh issue for what you are seeing.

@justbiaoliu In a broken checkout please do `grep -r NetworkMoude /path/to/ansible/` somewhere the is a typo.

**Comment by justbiaoliu** [comment]
@gundalow @kenwmitchell I'm sorry, I've spelled "NetworkModule" wrongly. Because the test equipment can not access to Internet, so the log is not copied over.   the title and log have been corrected.

**Comment by justbiaoliu** [comment]
@kenwmitchell  @gundalow 
Adding -a "_ansible_module_name=whatever" to test-module also works.  my test environment happen too.

**Comment by numone213** [comment]
@gundalow @kenwmitchell 
I have the same problem when i using hacking test.

**Comment by numone213** [comment]
when i use no_log=True to init NetworkModule, the hacking test will be ok, otherwise will be "AttributeError: 'NetworkModule' object has no attribute '_name'"

**Comment by kenwmitchell** [comment]
I think this was fixed in 461dcf8ce3edb0fafcbb608c846b9a68eddde019.

**Comment by gundalow** [comment]
Hi,
We believe this has been fixed, please reopen if you are still seeing the issue
